pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -178,22 +177,23 @@ func (l *listUsersQuery) ListUsers(
 	}()
 
 	go func() {
-		defer close(foundUsersCh)
 		internalRequest := fromListUsersRequest(req, &datastoreQueryCount)
-		if err := l.expand(cancellableCtx, internalRequest, foundUsersCh); err != nil {
-			expandErrCh <- err
-			return
-		}
+		err := l.expand(cancellableCtx, internalRequest, foundUsersCh)
+		close(foundUsersCh)
+		expandErrCh <- err
 	}()
 
-	select {
-	case err := <-expandErrCh:
-		telemetry.TraceError(span, err)
-		return nil, err
-	case <-done:
-		cancelContextIfMaxResultsMet()
-		break
+	err := <-expandErrCh
+	if err != nil {
+		if !errors.Is(err, context.DeadlineExceeded) {
+			telemetry.TraceError(span, err)
+			return nil, err
+		}
 	}
+
+	<-done
```

Comment:
Why did you remove the `select` statement? That will neglect the `max_results` setting.

Reply:
I thought I had the control flow covered correctly, but I was wrong. I've updated that channel's variable name so it's more indicative of it's relevance and have added it back in 👍

Reply:
Fixed 👍